src/array.js

```diff
@@ -6,9 +6,13 @@ if (!Array.from) {
     var toLength = function(it) {
       return it > 0 ? Math.min(toInteger(it), 0x1fffffffffffff) : 0; // pow(2, 53) - 1 == 9007199254740991
     };
-    var iterCall = function(iter, fn, a1, a2) {
+    var iterCall = function(fn, val, index, thisArg) {
       try {
-        fn(a1, a2);
+        if (thisArg) {
+          return fn.call(thisArg, val, index)
+        } else {
+          return fn(val, index);
+        }
       }
       catch (E) {
         if (typeof iter.return == 'function') iter.return();
```

Comment:
You removed the `iter` parameter and it can't be captured as it's not in scope here.